src/runtime/callbacks.js

```diff
@@ -172,10 +175,21 @@ export class Callbacks {
    * @return {boolean} Whether the callback has been found.
    */
   triggerPaymentResponse(responsePromise) {
-    return this.trigger_(
-      CallbackId.PAYMENT_RESPONSE,
-      responsePromise.then(res => res.clone())
+    this.paymentResponsePromise_ = responsePromise.then(
+      res => {
+        this.trigger_(
+          CallbackId.PAYMENT_RESPONSE,
+          Promise.resolve(res.clone())
+        );
+      },
+      reason => {
```

Comment:
This prevents the publisher from capturing the error too and dealing with it, right?

Reply:
They shouldn't have to catch a cancel error, they should be informed through our trigger flow cancel workflow.

If you mean the non-cancel errors, I would think this duplicates existing behavior.  If nothing was catching the error before it would effectively just be thrown to the next level?  I could be wrong though, my grasp of promises is not 100%.  I basically did this development because there were errors thrown when I closed the gPay popup which seems inappropriate.

Reply:
Hmm, I'm not 100% on promises either. Ok, if you're confident..

Reply:
If I'm reading the code correctly, though, doesn't this throw out cancel errors as well?

Reply:
If the user cancels (e.g. closes the popup window), the this.trigger code is never called.  It instead calls the part that starts with reason.  This is consuming an error thrown by the underlying promise.

In that catch, if it is the cancel error it just returns (basically canceling the error).  Otherwise, it throws the error so something higher level and be aware of it (ultimately it just goes to console).

Reply:
I must admit when I read your comment I wondered if you were correct.  It can be difficult to parse through how promises actually function...  I think both things are catching the same error because they operate on the same promise.  E.g.:

x = new Promise();
x.then(successFun, failFun);
x.then(successFun, failFun);
failFun will get called twice even if it doesn't throw the error.

To be certain, I just validated by testing this change locally.  I did so by adding this to DemoPaywalController in the swg-js/examples/sample-pub/demo.js

  this.subscriptions.setOnFlowCanceled(a => {
    debugger;
  });

And validated closing the gPay window hit that debugger point.

Reply:
Thanks for checking my dev though! I always appreciate multiple perspectives.

Reply:
Ah, okay. Thank you for checking!